pkg/cmd/cli/cmd/rsh.go

```diff
@@ -155,8 +155,9 @@ func (o *RshOptions) Validate() error {
 // Run starts a remote shell session on the server
 func (o *RshOptions) Run() error {
 	// Insert the TERM into the command to be run
-	term := fmt.Sprintf("TERM=%s", util.Env("TERM", "xterm"))
-	o.Command = append([]string{"env", term}, o.Command...)
-
+	if len(o.Command) == 1 && o.Command[0] == o.Executable {
+		termsh := fmt.Sprintf("TERM=%q %s", util.Env("TERM", "xterm"), o.Executable)
+		o.Command = append(o.Command, "-c", termsh)
```

Comment:
i agree if they pass a custom executable they are on their own, but if we tack a "-c" onto their custom executable and break it, we've screwed them and given them no real recourse (well, oc exec).  i'll change it back to only enter this block if it's /bin/sh.